File: Controller/component/utility/OrderTable.ts
/**
 * 注文に関するクラス
 */
export class OrderTable {
    /**
     * 注文内容
     */
    order: Flag[];
    serial: number;
    casherId: number;
    receptionTime: number;
    paymentType: number;
    price: number;

    /**
     * 受付番号
     */
    get Serial(): number {
        return this.serial
    }
    /**
     * 注文を受け取った従業員の固有番号
     */
    get CasherId(): number {
        return this.casherId;
    }
    /**
     * 注文受付時間
     */
    get ReceptionTime(): number {
        return this.receptionTime;
    }
    /**
     * 支払い方法
     */
    get PaymentType(): number {
        return this.paymentType;
    }


    constructor(serial: number, casherId: number, paymentType: number) {
        this.serial = serial;
        this.casherId = casherId;
        this.paymentType = paymentType;
        this.receptionTime = Date.now();
        this.order = [];
        this.price = 0;
    }

    Additem() {
        this.order.push(new Flag(0));
    }
}

export type TOrderTable = {
    // JSON から読み込むと Flag[] 型であることが保証されないため、自力でパースすること
    order: TFlag[];
    serial: number;
    casherId: number;
    receptionTime: number;
    paymentType: number;
}

export class Flag {
    flag: number = 0;

    constructor(flag: number) {
        this.flag = flag;
    }

    HasFlag(flag: number): boolean {
        return flag == (this.flag & flag);
    }

    AddFlag(flag: number) {
        this.flag |= flag;
    }

    RemoveFlag(flag: number) {
        this.flag &= ~flag;
    }
}

export type TFlag = {
    flag: number;
}

File: Display/Assets/Project/Scripts/Format/OrderTable.cs
using System;
using System.Linq;

public class OrderTable
{
    public int Serial { get; private set; }
    public int CasherId { get; private set; }
    public int PaymentType { get; private set; }
    public long ReceptionTime { get; private set; }
    public Flag[] Order { get; private set; }

    public OrderTable(OrderJsonOrder orderJson)
    {
        Serial = orderJson.serial;
        CasherId = orderJson.casherId;
        PaymentType = orderJson.paymentType;
        ReceptionTime = orderJson.receptionTime;
        Order = orderJson.order;
    }
}

[Serializable]
public class OrderJson
{
    public OrderJsonOrder[] data;
}

[Serializable]
public class OrderJsonOrder
{
    public int serial;
    public int casherId;
    public int paymentType;
    public long receptionTime;
    public Flag[] order;
}

[Serializable]
public class Flag
{
    public int flag;

    public bool HasFlag(int number)
    {
        return number == (flag & number);
    }
}